Tribler/Core/Modules/restapi/trustchain_endpoint.py

```diff
@@ -160,3 +167,191 @@ def render_GET(self, request):
+        - dataset
+            - Not given: TrustChain data
+            - "static": Static dummy data
+            - "random": Random dummy data
+            - otherwise: TrustChain data
+        - focus_node
+            - Not given: HTTP 400
+            - Non-String value: HTTP 400
+            - "self": TriblerChain Community public key
+            - otherwise: Passed data, albeit a string
+        - neighbor_level
+            - Not given: 1
+            - Non-Integer value: 1
+            - otherwise: Passed data, albeit an integer
+
+        The returned data will be in such format that the GUI component which visualizes this data can easily use it.
+        Although this data might not seem as formatted in a useful way to the human eye, this is done to accommodate as
+        little parsing effort at the GUI side.
+
+            **Example request**:
+
+            .. sourcecode:: none
+
+                curl -X GET 'http://localhost:8085/trustchain/network?dataset=static&focus_node=xyz&neighbor_level=1'
+
+            **Example response**:
+
+            .. sourcecode:: javascript
+
+                {
+                    "user_node": "abc",
+                    "focus_node": "xyz",
+                    "neighbor_level": 1,
+                    "nodes": [{
+                        "public_key": "xyz",
+                        "total_up": 12736457,
+                        "total_down": 1827364,
+                        "score": 0.0011,
+                        "total_neighbors": 1
+                    }, ...],
+                    "edges": [{
+                        "from": "xyz",
+                        "to": "xyz_n1",
+                        "amount": 12384
+                    }, ...]
+                }
+
+        :param request: the HTTP GET request which specifies the focus node and optionally the neighbor level
+        :return: the node data formatted in JSON
+        """
+        # This header is needed because this request is not made from the same host
+        request.setHeader('Access-Control-Allow-Origin', '*')
+
+        try:
+            tribler_chain_community = self.get_tribler_chain_community()
+        except OperationNotEnabledByConfigurationException as exc:
+            return TrustChainNetworkEndpoint.return_error(request, status_code=http.NOT_FOUND, message=exc.args)
+
+        if "dataset" in request.args:
+            self.use_dummy_data(request.args["dataset"][0], tribler_chain_community)
+
+        if "focus_node" not in request.args:
+            return TrustChainNetworkEndpoint.return_error(request, message="focus_node parameter missing")
+
+        focus_node = request.args["focus_node"][0]
+        if not focus_node:
+            return TrustChainNetworkEndpoint.return_error(request, message="focus_node parameter empty")
+
+        focus_node = self.get_focus_node(focus_node, tribler_chain_community)
+
+        if tribler_chain_community.persistence.dummy_setup:
```

Comment:
This will indeed be removed soon. See https://github.com/vandenheuvel/tribler/pull/358.